pkg/utils/prefixcacheindexer/tree.go

```diff
@@ -443,14 +442,14 @@ func (c *LPRadixCache) AddPrefix(tokens []int, model string, podName string) (*T
 
 	// Do insertion first
 	node, matchedTokens, unmatchedTokens := c.insertHelper(c.rootNode, tokens, tokens)
-	if node != nil {
+	if node != nil && podName != "" {
```

Comment:
unit tests did not fail. To productionize it unit and e2e tests are must. You can add them in follow up PR.